test/tracer_test.rb

```diff
@@ -142,4 +142,135 @@ def test_default_service
     assert_equal('foo-bar', tracer.default_service)
     tracer.default_service = old_service
   end
+
+  def test_active_span
+    tracer = get_test_tracer
+    span = tracer.trace('something')
+    assert_equal(span, tracer.active_span, 'current span is active')
+    assert_equal(false, tracer.active_span.finished?)
+  end
+
+  def test_trace_all_args
+    tracer = get_test_tracer
+    tracer.set_tags('env' => 'test', 'temp' => 'cool')
+
+    yesterday = Time.now.utc - 24 * 60 * 60
+    tracer.trace('op',
+                 service: 'special-service',
+                 resource: 'extra-resource',
+                 span_type: 'my-type',
+                 start_time: yesterday,
+                 tags: { 'tag1' => 'value1', 'tag2' => 'value2' }) do
+    end
+
+    spans = tracer.writer.spans()
+    assert_equal(1, spans.length)
+    span = spans[0]
+    assert_equal('special-service', span.service)
+    assert_equal('extra-resource', span.resource)
+    assert_equal('my-type', span.span_type)
+    assert_equal(yesterday, span.start_time)
+    assert_equal({ 'env' => 'test', 'temp' => 'cool', 'tag1' => 'value1', 'tag2' => 'value2' }, span.meta)
+  end
+
+  def test_start_span_all_args
+    tracer = get_test_tracer
+    tracer.set_tags('env' => 'test', 'temp' => 'cool')
+
+    yesterday = Time.now.utc - 24 * 60 * 60
+    span = tracer.start_span('op',
+                             service: 'special-service',
+                             resource: 'extra-resource',
+                             span_type: 'my-type',
+                             start_time: yesterday,
+                             tags: { 'tag1' => 'value1', 'tag2' => 'value2' })
+    span.finish
+
+    spans = tracer.writer.spans()
+    assert_equal(1, spans.length)
+    span = spans[0]
+    assert_equal('special-service', span.service)
+    assert_equal('extra-resource', span.resource)
+    assert_equal('my-type', span.span_type)
+    assert_equal(yesterday, span.start_time)
+    assert_equal({ 'env' => 'test', 'temp' => 'cool', 'tag1' => 'value1', 'tag2' => 'value2' }, span.meta)
+  end
+
+  def test_start_span_child_of_span
+    tracer = get_test_tracer
+
+    root = tracer.start_span('a')
+    root.finish
+
+    spans = tracer.writer.spans()
+    assert_equal(1, spans.length)
+    a = spans[0]
+
+    tracer.trace('b') do
+      span = tracer.start_span('c', child_of: root)
+      span.finish
+    end
+
+    spans = tracer.writer.spans()
+    assert_equal(2, spans.length)
+    b, c = spans
+
+    refute_equal(a.trace_id, b.trace_id, 'a and b do not belong to the same trace')
+    refute_equal(b.trace_id, c.trace_id, 'b and c do not belong to the same trace')
+    assert_equal(a.trace_id, c.trace_id, 'a and c belong to the same trace')
+    assert_equal(a.span_id, c.parent_id, 'a is the parent of c')
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  def test_start_span_child_of_context
+    # skipping this for Ruby 1.9, as it has some threading issues (segfaults on msgpack)
```

Comment:
Mhhh that's scary. What is the real problem here? are we breaking some support with Ruby 1.9? Also why msgpack is involved? it happens the same if we use the legacy mode (JSON?). Just to know what is happening here. Also if there is nothing we can do, I think it's better to expand the comment.

Nitpick - I think you can use that one instead of `return`:
```ruby
skip "Not available in Ruby 1.9" if RUBY_VERSION < '2.0.0'
```
otherwise the test exits with a success.